src/Features/Core/Portable/Completion/Providers/Snippets/AbstractSnippetCompletionProvider.cs

```diff
@@ -81,9 +80,9 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
                 return;
             }
 
-            var (strippedDocument, newPosition) = await GetDocumentWithoutInvokingTextAsync(document, position, cancellationToken).ConfigureAwait(false);
-
-            var snippets = await service.GetSnippetsAsync(strippedDocument, newPosition, cancellationToken).ConfigureAwait(false);
+            var syntaxContext = await context.GetSyntaxContextWithExistingSpeculativeModelAsync(document, cancellationToken).ConfigureAwait(false);
```

Comment:
This changes the content of the document passed in, just want to make sure this is fine.

Reply:
I couldn't find any case where it made a difference, since we still use the stripped document text for calculating the final change for a snippet.

Reply:
FYI @akhera99